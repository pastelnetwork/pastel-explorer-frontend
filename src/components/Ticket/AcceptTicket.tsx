import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';

import RouterLink from '@components/RouterLink/RouterLink';
import { formatNumber } from '@utils/helpers/formatNumbers/formatNumbers';
import { IAcceptTicket } from '@utils/types/ITransactions';
import { getCurrencyName } from '@utils/appInfo';
import * as ROUTES from '@utils/constants/routes';
import { formatFullDate } from '@utils/helpers/date/date';

import Signatures from './Signatures';
import * as Styles from './Ticket.styles';

interface IAcceptTicketProps {
  ticket: IAcceptTicket;
}

const AcceptTicket: React.FC<IAcceptTicketProps> = ({ ticket }) => {
  return (
    <Box>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>Pastel OpenAPI Ticket Version Number:</Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>{ticket.version}</Styles.TicketContent>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>Pastel ID:</Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>
            <RouterLink
              route={`${ROUTES.PASTEL_ID_DETAILS}/${ticket.pastelID}`}
              value={ticket.pastelID}
              title={ticket.pastelID}
              className="address-link"
            />
          </Styles.TicketContent>
        </Grid>
      </Grid>
      <Signatures signature={ticket.signature} />
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>Offer txid:</Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>
            {ticket.offer_txid ? (
              <RouterLink
                route={`${ROUTES.TRANSACTION_DETAILS}/${ticket.offer_txid}`}
                value={ticket.offer_txid}
                title={ticket.offer_txid}
                className="address-link"
              />
            ) : (
              'NA'
            )}
          </Styles.TicketContent>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>Price:</Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>
            {formatNumber(ticket.price)} {getCurrencyName()}
          </Styles.TicketContent>
        </Grid>
      </Grid>
      {ticket.transactionTime ? (
        <Grid container spacing={3}>
          <Grid item xs={4} sm={3} className="max-w-355">
            <Styles.TicketTitle>Timestamp:</Styles.TicketTitle>
          </Grid>
          <Grid item xs={8} sm={9}>
            <Styles.TicketContent>{formatFullDate(ticket.transactionTime)}</Styles.TicketContent>
          </Grid>
        </Grid>
      ) : null}
    </Box>
  );
};

export default AcceptTicket;
