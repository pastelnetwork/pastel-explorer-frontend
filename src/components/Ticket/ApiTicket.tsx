import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';

import { decode } from '@utils/helpers/ascii85';
import { IApiTicket } from '@utils/types/ITransactions';

import * as Styles from './Ticket.styles';

interface IApiTicketProps {
  apiTicket: string;
  actionType: string;
}

const ApiTicket: React.FC<IApiTicketProps> = ({ apiTicket, actionType }) => {
  if (!apiTicket) {
    return null;
  }
  const data = decode(apiTicket) as IApiTicket;
  return (
    <Box>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-370">
          <Styles.TicketTitle>Data hash:</Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>{data.data_hash || 'NA'}</Styles.TicketContent>
        </Grid>
      </Grid>
      {actionType === 'sense' ? (
        <>
          <Grid container spacing={3}>
            <Grid item xs={4} sm={3} className="max-w-370">
              <Styles.TicketTitle>Dd and fingerprints max:</Styles.TicketTitle>
            </Grid>
            <Grid item xs={8} sm={9}>
              <Styles.TicketContent>{data.dd_and_fingerprints_max}</Styles.TicketContent>
            </Grid>
          </Grid>
          <Grid container spacing={3}>
            <Grid item xs={4} sm={3} className="max-w-370">
              <Styles.TicketTitle>Dd and fingerprints ic:</Styles.TicketTitle>
            </Grid>
            <Grid item xs={8} sm={9}>
              <Styles.TicketContent>{data.dd_and_fingerprints_ic}</Styles.TicketContent>
            </Grid>
          </Grid>
          <Grid container spacing={3}>
            <Grid item xs={4} sm={3} className="max-w-370">
              <Styles.TicketTitle>Dd and fingerprints ids:</Styles.TicketTitle>
            </Grid>
            <Grid item xs={8} sm={9}>
              <Styles.TicketContent>
                {data.dd_and_fingerprints_ids ? data.dd_and_fingerprints_ids.join(', ') : 'NA'}
              </Styles.TicketContent>
            </Grid>
          </Grid>
        </>
      ) : null}
    </Box>
  );
};

export default ApiTicket;
