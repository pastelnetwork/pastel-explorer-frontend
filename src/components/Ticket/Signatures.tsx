import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';

import { ISignature } from '@utils/types/ITransactions';
import * as Styles from './Ticket.styles';

interface ISignaturesProps {
  signatures?: ISignature;
  signature?: string;
}

const Signatures: React.FC<ISignaturesProps> = ({ signatures, signature }) => {
  if (signature) {
    return (
      <Grid container spacing={3}>
        <Grid item xs={2}>
          <Styles.TicketTitle>Signature:</Styles.TicketTitle>
        </Grid>
        <Grid item xs={9}>
          <Styles.TicketContent className="break-all">{signature}</Styles.TicketContent>
        </Grid>
      </Grid>
    );
  }

  if (signatures) {
    return (
      <Box>
        <Grid container spacing={3}>
          <Grid item xs={2}>
            <Styles.TicketTitle>Principal PastelID:</Styles.TicketTitle>
          </Grid>
          <Grid item xs={9}>
            <Styles.TicketContent>{Object.keys(signatures.principal)[0]}</Styles.TicketContent>
          </Grid>
        </Grid>
        <Grid container spacing={3}>
          <Grid item xs={2}>
            <Styles.TicketTitle>Principal Signature:</Styles.TicketTitle>
          </Grid>
          <Grid item xs={9}>
            <Styles.TicketContent className="break-all">
              {Object.values(signatures.principal)[0]}
            </Styles.TicketContent>
          </Grid>
        </Grid>
        <Grid container spacing={3}>
          <Grid item xs={2}>
            <Styles.TicketTitle>Masternode 1 PastelID:</Styles.TicketTitle>
          </Grid>
          <Grid item xs={9}>
            <Styles.TicketContent className="break-all">
              {Object.keys(signatures.mn1)[0]}
            </Styles.TicketContent>
          </Grid>
        </Grid>
        <Grid container spacing={3}>
          <Grid item xs={2}>
            <Styles.TicketTitle>Masternode 1 Signature:</Styles.TicketTitle>
          </Grid>
          <Grid item xs={9}>
            <Styles.TicketContent className="break-all">
              {Object.values(signatures.mn1)[0]}
            </Styles.TicketContent>
          </Grid>
        </Grid>
        <Grid container spacing={3}>
          <Grid item xs={2}>
            <Styles.TicketTitle>Masternode 2 PastelID:</Styles.TicketTitle>
          </Grid>
          <Grid item xs={9}>
            <Styles.TicketContent className="break-all">
              {Object.keys(signatures.mn2)[0]}
            </Styles.TicketContent>
          </Grid>
        </Grid>
        <Grid container spacing={3}>
          <Grid item xs={2}>
            <Styles.TicketTitle>Masternode 2 Signature:</Styles.TicketTitle>
          </Grid>
          <Grid item xs={9}>
            <Styles.TicketContent className="break-all">
              {Object.values(signatures.mn2)[0]}
            </Styles.TicketContent>
          </Grid>
        </Grid>
        <Grid container spacing={3}>
          <Grid item xs={2}>
            <Styles.TicketTitle>Masternode 3 PastelID:</Styles.TicketTitle>
          </Grid>
          <Grid item xs={9}>
            <Styles.TicketContent className="break-all">
              {Object.keys(signatures.mn3)[0]}
            </Styles.TicketContent>
          </Grid>
        </Grid>
        <Grid container spacing={3}>
          <Grid item xs={2}>
            <Styles.TicketTitle>Masternode 3 Signature:</Styles.TicketTitle>
          </Grid>
          <Grid item xs={9}>
            <Styles.TicketContent className="break-all">
              {Object.values(signatures.mn3)[0]}
            </Styles.TicketContent>
          </Grid>
        </Grid>
      </Box>
    );
  }

  return null;
};

export default Signatures;
