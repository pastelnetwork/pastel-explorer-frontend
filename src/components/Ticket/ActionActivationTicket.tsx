import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';

import RouterLink from '@components/RouterLink/RouterLink';
import { formatNumber } from '@utils/helpers/formatNumbers/formatNumbers';
import { IActionActivationTicket } from '@utils/types/ITransactions';
import { getCurrencyName } from '@utils/appInfo';
import * as ROUTES from '@utils/constants/routes';
import { formatFullDate } from '@utils/helpers/date/date';

import Signatures from './Signatures';
import * as Styles from './Ticket.styles';

interface IActionActivationTicketProps {
  ticket: IActionActivationTicket;
}

const ActionActivationTicket: React.FC<IActionActivationTicketProps> = ({ ticket }) => {
  return (
    <Box>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>Pastel Block Height When Ticket Registered:</Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>
            {ticket.called_at ? (
              <RouterLink
                route={`${ROUTES.BLOCK_DETAILS}/${ticket.called_at}`}
                value={ticket.called_at}
                title={ticket.called_at?.toString()}
                className="address-link"
              />
            ) : (
              'NA'
            )}
          </Styles.TicketContent>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>Pastel OpenAPI Ticket Version Number:</Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>{ticket.version}</Styles.TicketContent>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>Pastel ID:</Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>
            <RouterLink
              route={`${ROUTES.PASTEL_ID_DETAILS}/${ticket.pastelID}`}
              value={ticket.pastelID}
              title={ticket.pastelID}
              className="address-link"
            />
          </Styles.TicketContent>
        </Grid>
      </Grid>
      <Signatures signature={ticket.signature} />
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>Reg txid:</Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>
            {ticket.reg_txid ? (
              <RouterLink
                route={`${ROUTES.TRANSACTION_DETAILS}/${ticket.reg_txid}`}
                value={ticket.reg_txid}
                title={ticket.reg_txid}
                className="address-link"
              />
            ) : (
              'NA'
            )}
          </Styles.TicketContent>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>
            Total Cost in PSL to Register Ticket on Blockchain:
          </Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>
            {formatNumber(ticket.storage_fee)} {getCurrencyName()}
          </Styles.TicketContent>
        </Grid>
      </Grid>
      {ticket.transactionTime ? (
        <Grid container spacing={3}>
          <Grid item xs={4} sm={3} className="max-w-355">
            <Styles.TicketTitle>Timestamp:</Styles.TicketTitle>
          </Grid>
          <Grid item xs={8} sm={9}>
            <Styles.TicketContent>{formatFullDate(ticket.transactionTime)}</Styles.TicketContent>
          </Grid>
        </Grid>
      ) : null}
    </Box>
  );
};

export default ActionActivationTicket;
