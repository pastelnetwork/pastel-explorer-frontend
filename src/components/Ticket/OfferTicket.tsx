import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';

import RouterLink, { ExternalLink } from '@components/RouterLink/RouterLink';
import { formatNumber } from '@utils/helpers/formatNumbers/formatNumbers';
import { IOfferTicket } from '@utils/types/ITransactions';
import * as ROUTES from '@utils/constants/routes';
import { getCurrencyName } from '@utils/appInfo';
import { formatFullDate } from '@utils/helpers/date/date';
import { translate } from '@utils/helpers/i18n';
import CopyButton from '@components/CopyButton/CopyButton';

import noImagePlaceholder from '@assets/images/no-image-placeholder.svg';

import { useStorageFee } from './Ticket.helpers';
import Signatures from './Signatures';
import * as Styles from './Ticket.styles';

interface IOfferTicketProps {
  ticket: IOfferTicket;
  variant?: string;
}

const OfferTicket: React.FC<IOfferTicketProps> = ({ ticket, variant }) => {
  const { storageFee } = useStorageFee(ticket.asked_price);
  const getImage = () => {
    let image = noImagePlaceholder;
    if (ticket.image) {
      image =
        ticket.otherData.offerType === 'nft-offer'
          ? ticket.image
          : `data:image/jpeg;base64,${ticket.image}`;
    }
    return (
      <img
        src={image}
        alt={ticket.otherData.txId}
        className={`nft-image ${!ticket.image ? 'placeholder' : ''}`}
      />
    );
  };
  if (variant === 'transaction') {
    const renderDetailLink = () => {
      if (ticket.otherData.offerType === 'sense') {
        return (
          <ExternalLink
            href={`${ROUTES.SENSE_DETAILS}?txid=${ticket.otherData.ticketId}`}
            value={getImage()}
          />
        );
      }
      if (ticket.otherData.ticketType === 'cascade') {
        return (
          <ExternalLink
            href={`${ROUTES.CASCADE_DETAILS}?txid=${ticket.otherData.ticketId}`}
            value={getImage()}
          />
        );
      }

      return (
        <ExternalLink
          href={`${ROUTES.NFT_DETAILS}?txid=${ticket.otherData.ticketId}#offers`}
          value={getImage()}
        />
      );
    };
    const getOfferType = () => {
      switch (ticket.otherData.offerType) {
        case 'sense':
          return translate('components.ticket.offerTicket.sense');
        case 'cascade':
          return translate('components.ticket.offerTicket.cascade');
        default:
          return translate('components.ticket.offerTicket.nft');
      }
    };

    return (
      <Styles.OfferWrapper>
        <Grid container spacing={3}>
          <Grid item xs={12} md={3} className="max-w-355">
            {renderDetailLink()}
          </Grid>
          <Grid item xs={12} md={9}>
            <Grid container spacing={3} className="mb-sm-8">
              <Grid item xs={12} md={3} className="max-w-355">
                <Styles.TicketTitle>
                  {translate('components.ticket.offerTicket.pastelID')}
                </Styles.TicketTitle>
              </Grid>
              <Grid item xs={12} md={9}>
                <Styles.TicketContent>
                  {ticket.pastelID && ticket.pastelID !== 'not defined' ? (
                    <RouterLink
                      route={`${ROUTES.PASTEL_ID_DETAILS}/${ticket.pastelID}`}
                      value={ticket.pastelID}
                      title={ticket.pastelID}
                      className="address-link"
                    />
                  ) : (
                    translate('common.na')
                  )}
                </Styles.TicketContent>
              </Grid>
            </Grid>
            <Grid container spacing={3} className="mb-sm-8">
              <Grid item xs={12} md={3} className="max-w-355">
                <Styles.TicketTitle>
                  {translate('components.ticket.offerTicket.copyNumber')}
                </Styles.TicketTitle>
              </Grid>
              <Grid item xs={12} sm={9}>
                <Styles.TicketContent>{ticket.copy_number}</Styles.TicketContent>
              </Grid>
            </Grid>
            <Grid container spacing={3} className="mb-sm-8">
              <Grid item xs={12} sm={3} className="max-w-355">
                <Styles.TicketTitle>
                  {translate('components.ticket.offerTicket.askedPrice')}
                </Styles.TicketTitle>
              </Grid>
              <Grid item xs={12} sm={9}>
                <Styles.TicketContent>
                  {formatNumber(ticket.asked_price)} {getCurrencyName()} {storageFee}
                </Styles.TicketContent>
              </Grid>
            </Grid>
            <Grid container spacing={3} className="mb-sm-8">
              <Grid item xs={12} sm={3} className="max-w-355">
                <Styles.TicketTitle>
                  {translate('components.ticket.offerTicket.itemTxId')}
                </Styles.TicketTitle>
              </Grid>
              <Grid item xs={12} sm={9}>
                <Styles.TicketContent>
                  <CopyButton copyText={ticket?.item_txid || ticket?.nft_txid} />
                  <RouterLink
                    route={`${ROUTES.TRANSACTION_DETAILS}/${ticket?.item_txid || ticket?.nft_txid}`}
                    value={ticket?.item_txid || ticket?.nft_txid}
                    title={ticket?.item_txid || ticket?.nft_txid}
                    className="address-link"
                  />
                </Styles.TicketContent>
              </Grid>
            </Grid>
            <div className="driver" />
            <Grid container spacing={3} className="mb-sm-8">
              <Grid item xs={12} sm={3} className="max-w-355">
                <Styles.TicketTitle>
                  {translate('components.ticket.offerTicket.type')}:
                </Styles.TicketTitle>
              </Grid>
              <Grid item xs={12} sm={9}>
                <Styles.TicketContent>{getOfferType()}</Styles.TicketContent>
              </Grid>
            </Grid>
            <Grid container spacing={3} className="mb-sm-8">
              <Grid item xs={12} sm={3} className="max-w-355">
                <Styles.TicketTitle>
                  {translate('components.ticket.offerTicket.version')}
                </Styles.TicketTitle>
              </Grid>
              <Grid item xs={12} sm={9}>
                <Styles.TicketContent>{ticket.version}</Styles.TicketContent>
              </Grid>
            </Grid>
            <Grid container spacing={3} className="mb-sm-8">
              <Grid item xs={12} sm={3} className="max-w-355">
                <Styles.TicketTitle>
                  {translate('components.ticket.offerTicket.validAfter')}
                </Styles.TicketTitle>
              </Grid>
              <Grid item xs={12} sm={9}>
                <Styles.TicketContent>
                  {ticket.valid_after ? (
                    <RouterLink
                      route={`${ROUTES.BLOCK_DETAILS}/${ticket.valid_after}`}
                      value={ticket.valid_after}
                      title={ticket.valid_after?.toString()}
                      className="address-link"
                    />
                  ) : (
                    ticket.valid_after
                  )}
                </Styles.TicketContent>
              </Grid>
            </Grid>
            <Grid container spacing={3} className="mb-sm-8">
              <Grid item xs={12} sm={3} className="max-w-355">
                <Styles.TicketTitle>
                  {translate('components.ticket.offerTicket.validBefore')}
                </Styles.TicketTitle>
              </Grid>
              <Grid item xs={12} sm={9}>
                <Styles.TicketContent>
                  {ticket.valid_before ? (
                    <RouterLink
                      route={`${ROUTES.BLOCK_DETAILS}/${ticket.valid_before}`}
                      value={ticket.valid_before}
                      title={ticket.valid_before?.toString()}
                      className="address-link"
                    />
                  ) : (
                    ticket.valid_before
                  )}
                </Styles.TicketContent>
              </Grid>
            </Grid>
            <Grid container spacing={3} className="mb-sm-8">
              <Grid item xs={12} sm={3} className="max-w-355">
                <Styles.TicketTitle>
                  {translate('components.ticket.offerTicket.lockedRecipient')}
                </Styles.TicketTitle>
              </Grid>
              <Grid item xs={12} sm={9}>
                <Styles.TicketContent>
                  {ticket.locked_recipient !== 'not defined' ? (
                    <RouterLink
                      route={`${ROUTES.PASTEL_ID_DETAILS}/${ticket.locked_recipient}`}
                      value={ticket.locked_recipient}
                      title={ticket.locked_recipient}
                      className="address-link"
                    />
                  ) : (
                    ticket.locked_recipient
                  )}
                </Styles.TicketContent>
              </Grid>
            </Grid>
            <Signatures signature={ticket.signature} variant={variant} />
            {ticket.transactionTime ? (
              <Grid container spacing={3}>
                <Grid item xs={12} sm={3} className="max-w-355">
                  <Styles.TicketTitle>
                    {translate('components.ticket.offerTicket.timestamp')}
                  </Styles.TicketTitle>
                </Grid>
                <Grid item xs={12} sm={9}>
                  <Styles.TicketContent>
                    {formatFullDate(ticket.transactionTime, { dayName: false })}
                  </Styles.TicketContent>
                </Grid>
              </Grid>
            ) : null}
          </Grid>
        </Grid>
      </Styles.OfferWrapper>
    );
  }
  return (
    <Box>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>
            {translate('components.ticket.offerTicket.version')}
          </Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>{ticket.version}</Styles.TicketContent>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>
            {translate('components.ticket.offerTicket.pastelID')}
          </Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>
            {ticket.pastelID && ticket.pastelID !== 'not defined' ? (
              <RouterLink
                route={`${ROUTES.PASTEL_ID_DETAILS}/${ticket.pastelID}`}
                value={ticket.pastelID}
                title={ticket.pastelID}
                className="address-link"
              />
            ) : (
              translate('common.na')
            )}
          </Styles.TicketContent>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>
            {translate('components.ticket.offerTicket.copyNumber')}
          </Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>{ticket.copy_number}</Styles.TicketContent>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>
            {translate('components.ticket.offerTicket.askedPrice')}
          </Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>
            {formatNumber(ticket.asked_price)} {getCurrencyName()} {storageFee}
          </Styles.TicketContent>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>
            {translate('components.ticket.offerTicket.itemTxId')}
          </Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>
            <CopyButton copyText={ticket?.item_txid || ticket?.nft_txid} />
            <RouterLink
              route={`${ROUTES.TRANSACTION_DETAILS}/${ticket?.item_txid || ticket?.nft_txid}`}
              value={ticket?.item_txid || ticket?.nft_txid}
              title={ticket?.item_txid || ticket?.nft_txid}
              className="address-link"
            />
          </Styles.TicketContent>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>
            {translate('components.ticket.offerTicket.validAfter')}
          </Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>
            {ticket.valid_after ? (
              <RouterLink
                route={`${ROUTES.BLOCK_DETAILS}/${ticket.valid_after}`}
                value={ticket.valid_after}
                title={ticket.valid_after?.toString()}
                className="address-link"
              />
            ) : (
              ticket.valid_after
            )}
          </Styles.TicketContent>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>
            {translate('components.ticket.offerTicket.validBefore')}
          </Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>
            {ticket.valid_before ? (
              <RouterLink
                route={`${ROUTES.BLOCK_DETAILS}/${ticket.valid_before}`}
                value={ticket.valid_before}
                title={ticket.valid_before?.toString()}
                className="address-link"
              />
            ) : (
              ticket.valid_before
            )}
          </Styles.TicketContent>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item xs={4} sm={3} className="max-w-355">
          <Styles.TicketTitle>
            {translate('components.ticket.offerTicket.lockedRecipient')}
          </Styles.TicketTitle>
        </Grid>
        <Grid item xs={8} sm={9}>
          <Styles.TicketContent>{ticket.locked_recipient}</Styles.TicketContent>
        </Grid>
      </Grid>
      <Signatures signature={ticket.signature} />
      {ticket.transactionTime ? (
        <Grid container spacing={3}>
          <Grid item xs={4} sm={3} className="max-w-355">
            <Styles.TicketTitle>
              {translate('components.ticket.offerTicket.timestamp')}
            </Styles.TicketTitle>
          </Grid>
          <Grid item xs={8} sm={9}>
            <Styles.TicketContent>
              {formatFullDate(ticket.transactionTime, { dayName: false })}
            </Styles.TicketContent>
          </Grid>
        </Grid>
      ) : null}
    </Box>
  );
};

export default OfferTicket;
