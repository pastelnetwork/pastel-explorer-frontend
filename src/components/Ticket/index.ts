import { TTicketType } from '@utils/types/ITransactions';
import { translate } from '@utils/helpers/i18n';

import PastelIDRegistrationTicket from './PastelIDRegistrationTicket';
import UserNameChangeTicket from './UserNameChangeTicket';
import NFTRegistrationTicket from './NFTRegistrationTicket';
import NFTActivationTicket from './NFTActivationTicket';
import NFTCollectionRegistrationTicket from './NFTCollectionRegistrationTicket';
import NFTCollectionActivationTicket from './NFTCollectionActivationTicket';
import NFTRoyaltyTicket from './NFTRoyaltyTicket';
import ActionActivationTicket from './ActionActivationTicket';
import ActionRegistrationTicket from './ActionRegistrationTicket';
import OfferTicket from './OfferTicket';
import AcceptTicket from './AcceptTicket';
import TransferTicket from './TransferTicket';

const getTicketTitle = (type: TTicketType, itemType = '') => {
  switch (type) {
    case 'pastelid':
      return translate('components.ticket.ticketsTitle.pastelid');
    case 'username-change':
      return translate('components.ticket.ticketsTitle.usernameChange');
    case 'nft-reg':
      return translate('components.ticket.ticketsTitle.nftReg');
    case 'nft-act':
      return translate('components.ticket.ticketsTitle.nftAct');
    case 'collection-reg':
      return itemType === 'sense'
        ? translate('components.ticket.ticketsTitle.senseCollectionRegistrationTicket')
        : translate('components.ticket.ticketsTitle.nftCollectionReg');
    case 'collection-act':
      return translate('components.ticket.ticketsTitle.nftCollectionAct');
    case 'nft-royalty':
      return translate('components.ticket.ticketsTitle.nftRoyalty');
    case 'action-reg':
      return translate('components.ticket.ticketsTitle.actionReg');
    case 'action-act':
      return translate('components.ticket.ticketsTitle.actionAct');
    case 'offer':
      return translate('components.ticket.ticketsTitle.offer');
    case 'accept':
      return translate('components.ticket.ticketsTitle.accept');
    case 'transfer':
      return translate('components.ticket.ticketsTitle.transfer');
    default:
      return '';
  }
};

export {
  PastelIDRegistrationTicket,
  UserNameChangeTicket,
  NFTRegistrationTicket,
  NFTActivationTicket,
  NFTCollectionRegistrationTicket,
  NFTCollectionActivationTicket,
  NFTRoyaltyTicket,
  ActionActivationTicket,
  ActionRegistrationTicket,
  OfferTicket,
  AcceptTicket,
  TransferTicket,
  getTicketTitle,
};
