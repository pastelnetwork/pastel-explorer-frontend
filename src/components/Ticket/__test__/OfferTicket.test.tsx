import { shallow } from 'enzyme';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import 'jest-styled-components';

import { MyMockType } from '@utils/types/MockType';
import RouterLink from '../../RouterLink/RouterLink';
import i18next from '../../../utils/helpers/i18n';
import OfferTicket from '../OfferTicket';
import Signatures from '../Signatures';
import * as Styles from '../Ticket.styles';

jest.mock('i18next-http-backend');
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () =>
          new Promise(() => {
            // noop
          }),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {
      // noop
    },
  },
}));
i18next.t = jest.fn().mockImplementation((...arg) => {
  return arg[0];
}) as MyMockType;

describe('components/OfferTicket', () => {
  const ticket = {
    type: 'offer',
    version: 0,
    pastelID:
      'jXac5N8FdQWM354v9zpk2rEqfuWGXfeti2EuV44AgFsxiJWHzdxhprjD68cgr3H8qUJ5gydmXjYKqQ9ubF131s',
    item_txid: '6b43be16291af3b546fa4309d42497fc01927f1898eebb09e1bc9541e4d8f582',
    nft_txid: '7b43be16291af3b546fa4309d42497fc01927f1898eebb09e1bc9541e4d8f582',
    copy_number: 1,
    asked_price: 1,
    valid_after: 0,
    valid_before: 0,
    locked_recipient:
      'jXZqgvtqGERZuAbNkEXV1WXonpay98zUUgTk4AWwx2vSJq42vBxwyvwQZ36dVEAkBVB6sYqPh46SvktveCA93L',
    signature:
      '884cb82675f4eca59a7fbdac7bb093884bc9cca8749bbe43b81b2e5871576b14d1b3fd157fa8a8c2e787cc44ef451e9cdb4568f8c1a949d100e0af5738abbf4cdc85e06614c194d2ce3943823fb5ffb4fbf210330442f94067aafcf7f240108feeca31f19c1c0f465919b7c8ff59a9ab2800',
    image: '',
    transactionTime: 1675206868000,
    otherData: {
      offerType: '',
      txId: '',
      regTxId: '',
      ticketType: '',
      ticketId: '',
    },
  };
  const wrapper = shallow(<OfferTicket ticket={ticket} />);

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <Grid>', () => {
    expect(wrapper.find(Grid).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Box>', () => {
    expect(wrapper.find(Box).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <RouterLink>', () => {
    expect(wrapper.find(RouterLink).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Signatures>', () => {
    expect(wrapper.find(Signatures).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.TicketTitle>', () => {
    expect(wrapper.find(Styles.TicketTitle).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.TicketContent>', () => {
    expect(wrapper.find(Styles.TicketContent).length).toBeGreaterThanOrEqual(1);
  });
});
