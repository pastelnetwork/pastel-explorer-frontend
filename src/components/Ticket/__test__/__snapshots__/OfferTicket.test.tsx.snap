// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/OfferTicket renders correctly 1`] = `ShallowWrapper {}`;
